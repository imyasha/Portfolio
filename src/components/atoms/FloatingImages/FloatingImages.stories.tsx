import { FloatingImages, FloatingImagesProps } from './FloatingImages';

export default {
	title: 'atoms/FloatingImages',
	component: FloatingImages,
};

export const FloatingImagesStory = (args: FloatingImagesProps) => (
	<FloatingImages {...args} />
);

FloatingImagesStory.storyName = 'FloatingImages';
FloatingImagesStory.args = {
	mainImage: '/some-img.png',
	topFloatingImage: '/top-image.png',
	bottomFloatingImage: '/bottom-image.png',
	altText: 'Some image',
};
