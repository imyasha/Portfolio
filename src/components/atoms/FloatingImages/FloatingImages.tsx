import Image from 'next/image';

export interface FloatingImagesProps {
	altText: string;
	mainImage: string;
	topFloatingImage: string;
	bottomFloatingImage: string;
}

const FloatingImages = ({
	altText,
	mainImage,
	topFloatingImage,
	bottomFloatingImage,
}: FloatingImagesProps) => {
	return (
		<div className="relative inline-block w-auto">
			<Image src={mainImage} width={475} height={275} alt={altText} />
			<div className="absolute w-1/2 top-0 left-12 md:left-20 z-30 animate-hover">
				<Image src={topFloatingImage} width={330} height={210} alt={altText} />
			</div>
			<div className="absolute w-1/2 bottom-0 right-12 md:right-20 z-30 animate-hover animation-delay">
				<Image
					src={bottomFloatingImage}
					width={330}
					height={210}
					alt={altText}
				/>
			</div>
		</div>
	);
};

export { FloatingImages };
