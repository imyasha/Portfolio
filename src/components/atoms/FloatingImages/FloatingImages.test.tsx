import { FloatingImages } from '../FloatingImages';
import { render } from '@testing-library/react';

describe('FloatingImages', () => {
	it('renders correctly', () => {
		const { container } = render(
			<FloatingImages
				mainImage="/some-img.png"
				topFloatingImage="/top-image.png"
				bottomFloatingImage="/bottom-image.png"
				altText="Some image"
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
