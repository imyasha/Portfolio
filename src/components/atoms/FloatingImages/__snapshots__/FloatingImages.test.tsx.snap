// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingImages renders correctly 1`] = `
<div>
  <div
    class="relative inline-block w-auto"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27475%27%20height=%27275%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Some image"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <div
      class="absolute w-1/2 top-0 left-12 md:left-20 z-30 animate-hover"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27330%27%20height=%27210%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Some image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="absolute w-1/2 bottom-0 right-12 md:right-20 z-30 animate-hover animation-delay"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27330%27%20height=%27210%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Some image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</div>
`;
